distributed/worker.py

```diff
@@ -2264,6 +2264,14 @@ async def execute(self, key: str, *, stimulus_id: str) -> StateMachineEvent:
                 )
 
             self.active_keys.add(key)
+            # Propagate span (see distributed.spans). This is useful when spawning
+            # more tasks using worker_client() and for logging.
+            if "span" in ts.annotations:
+                span_ctx = dask.annotate(span=ts.annotations["span"])
+                span_ctx.__enter__()
+            else:
+                span_ctx = None
```

Comment:
nit: Setting `span_ctx = contextlib.nullcontext()` here and wrapping the following code in `with span_ctx` instead of explicitly entering and later exiting in the `finally` clause might be simpler for future maintenance. Feel free to ignore.

Reply:
Done